lstchain/reco/utils.py

```diff
@@ -201,17 +201,18 @@ def get_event_pos_in_camera(event, tel):
     -------
     (x, y) (float, float): position in the camera
     """
-    array_pointing = HorizonFrame(alt=event.mcheader.run_array_direction[1],
+    array_pointing = AltAz(alt=event.mcheader.run_array_direction[1],
                                   az=event.mcheader.run_array_direction[0])
-    event_direction = HorizonFrame(alt=event.mc.alt.to(u.rad),
+    event_direction = AltAz(alt=event.mc.alt.to(u.rad),
                                    az=event.mc.az.to(u.rad))
 
-    nom_frame = NominalFrame(array_direction=array_pointing,
-                         pointing_direction=array_pointing)
-
-    event_dir_nom = event_direction.transform_to(nom_frame)
     focal = tel.optics.equivalent_focal_length
-    return focal * (event_dir_nom.x.to(u.rad).value, event_dir_nom.y.to(u.rad).value)
+
+    camera_pos = sky_to_camera(event_direction.alt, event_direction.az,
```

Comment:
you can transform directly from AltAz to CameraFrame using the `transform_to()` method.
So you have your coordinates in the `camera_frame` and you can just do (provided `array_pointing` is in AltAz):
`camera_pos = array_pointing.transform_to(camera_frame)`